mailbox/api/src/main/java/org/apache/james/mailbox/model/Attachment.java

```diff
@@ -36,35 +34,41 @@ public static Builder builder() {
     }
 
     public static class Builder {
-
         private AttachmentId attachmentId;
-        private byte[] bytes;
         private String type;
+        private Long size;
 
         public Builder attachmentId(AttachmentId attachmentId) {
             Preconditions.checkArgument(attachmentId != null);
             this.attachmentId = attachmentId;
             return this;
         }
 
-        public Builder bytes(byte[] bytes) {
-            Preconditions.checkArgument(bytes != null);
-            this.bytes = bytes;
-            return this;
-        }
-
         public Builder type(String type) {
             Preconditions.checkArgument(!Strings.isNullOrEmpty(type));
             this.type = type;
             return this;
         }
 
+        public Builder size(long size) {
+            Preconditions.checkArgument(size >= 0, "'size' needs to be positive");
+            this.size = size;
+            return this;
+        }
+
         public Attachment build() {
-            Preconditions.checkState(bytes != null, "'bytes' is mandatory");
             Preconditions.checkState(type != null, "'type' is mandatory");
+            Preconditions.checkState(size != null, "'size' is mandatory");
             AttachmentId builtAttachmentId = attachmentId();
             Preconditions.checkState(builtAttachmentId != null, "'attachmentId' is mandatory");
-            return new Attachment(bytes, builtAttachmentId, type, size());
+
+            return new Attachment(builtAttachmentId, type, size);
+        }
+
+        public Attachment.WithBytes buildWithBytes(byte[] bytes) {
```

Comment:
why should we need that _at all_ when one can call `retrieveContent` at anytime?

Reply:
Write path. See MessageParser.